### PR TITLE
using aux::array in metrics array and aux::vector in session_stats_metrics

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: src/session_stats.cpp [hunk 1]
```diff
@@ -44,8 +47,8 @@ namespace libtorrent
 	};
 
 #define METRIC(category, name) { #category "." #name, counters:: name },
-	static const stats_metric_impl metrics[] =
-	{
+	aux::array<stats_metric_impl, counters::num_counters> const metrics
```
Comment (arvidn) [inline src/session_stats.cpp]: if you remove ``static`` (which seems reasonable to do), you need to put it in the anonymous namespace

Comment (aldenml) [inline src/session_stats.cpp]: right

File: src/session_stats.cpp [hunk 2]
```diff
@@ -545,15 +548,14 @@ namespace libtorrent
...
+	}});
 #undef METRIC
 
 	std::vector<stats_metric> session_stats_metrics()
 	{
-		std::vector<stats_metric> stats;
-		int const num = sizeof(metrics) / sizeof(metrics[0]);
-		stats.resize(num);
-		for (std::size_t i = 0; i < num; ++i)
+		aux::vector<stats_metric> stats;
```
Comment (arvidn) [inline src/session_stats.cpp]: will the return statement call the move constructor of ``std::vector``, despite slicing?

**Comment by arvidn** [comment]
lgtm

File: src/session_stats.cpp (hunk 2 above)
Comment (aldenml) [inline src/session_stats.cpp]: I don't know, how can I verify this? (since this is not the only place when `aux::*` is in the return for an `std::*` type)

Comment (aldenml) [inline src/session_stats.cpp]: I kind of assumed static cast to a base type

Comment (arvidn) [inline src/session_stats.cpp]: yeah, it's probably fine. I imagine ``std::vector<stats_metric>(std::vector&& rhs);`` will be invoked, and ``aux::vector<stats_metric>`` is cast to a reference to its baseclass.

**Comment by arvidn** [comment]
lgtm

**Pull request merged**